src/sonic-yang-mgmt/tests/libyang-python-tests/sample-yang-models/sonic-acl.yang

```diff
@@ -0,0 +1,274 @@
+module sonic-acl {
+
+	yang-version 1.1;
+
+	namespace "http://github.com/Azure/sonic-acl";
+	prefix acl;
+
+	import ietf-yang-types {
+		prefix yang;
+	}
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-head {
+		prefix head;
+		revision-date 2019-07-01;
+	}
+
+	import sonic-port {
+		prefix port;
+		revision-date 2019-07-01;
+	}
+
+	import sonic-portchannel {
+		prefix lag;
+		revision-date 2019-07-01;
+	}
+
+	organization "Linkedin Corporation";
+
+	contact "lnos_coders@linkedin.com";
+
+	description "ACL YANG Module for SONiC OS";
+
+	revision 2019-07-01 {
+		description "First Revision";
+	}
+
+	container sonic-acl {
+
+		container ACL_RULE {
+
+			description "ACL_RULE part of config_db.json";
+
+			list ACL_RULE_LIST {
+
+				key "ACL_TABLE_NAME RULE_NAME";
+
+				leaf ACL_TABLE_NAME {
+					type leafref {
+						path "/acl:sonic-acl/acl:ACL_TABLE/acl:ACL_TABLE_LIST/acl:ACL_TABLE_NAME";
+					}
+				}
+
+				leaf RULE_NAME {
+					type string {
+						length 1..255;
+					}
+				}
+
+				leaf PACKET_ACTION {
+					type head:packet_action;
+				}
+
+				leaf IP_TYPE {
+					type head:ip_type;
+				}
+
+				leaf PRIORITY {
+					type uint32 {
+						range 0..999999;
+					}
+				}
+
+				choice ip_prefix {
+
+					case ip4_prefix {
+						when "boolean(IP_TYPE[.='ANY' or .='IP' or .='IPV4' or .='IPV4ANY' or .='ARP'])";
```

Comment:
How does this mandate, when IP_TYPE is IP, a  non-empty ip-prefix is given ?

Reply:
These just sample YANG models for testing, not real ones [Actual YANG models were in PR 3730]. I will cherry-pick Ping`s fix which changes the file name from sonic-scl.yang to test-sonic-acl.yang. That fix is going under test. 

To answer your ques:, leaf type is inet:ip-prefix, which has a pattern condition, so empty string does not match the pattern.
]\